File: acmp.ru/[0501] - [0600]/[0577] Таблица умножения.cpp
#include <iostream>
using namespace std;


int ans[10] = {0};


int main() {
    int n, m;
    cin >> n >> m;

    for (int i = 1; i <= n; ++i) {
        for (int j = 1; j <= m; ++j) {
            int x = i * j;
            while (x) {
                ++ans[x % 10];
                x /= 10;
            }
        }
    }

    for (int i : ans)
        cout << i << '\n';

    return 0;
}

File: acmp.ru/[0501] - [0600]/[0577] Таблица умножения.py
n, m = map(int, input().split())
ans = [0] * 10

for i in range(1, n + 1):
    for j in range(1, m + 1):
        for c in str(i * j):
            ans[int(c)] += 1

print(*ans, sep='\n')
